src/controllers/informationController.js

```diff
@@ -1,49 +1,59 @@
-const mongoose = require('mongoose');
+// const mongoose = require('mongoose');
 // const userProfile = require('../models/userProfile');
 // const hasPermission = require('../utilities/permissions');
 const escapeRegex = require('../utilities/escapeRegex');
-const cache = require('../utilities/nodeCache')();
+const cacheClosure = require('../utilities/nodeCache');
 
 const informationController = function (Information) {
+  const cache = cacheClosure();
   const getInformations = function (req, res) {
     // return all informations if cache is available
-    if (cache.hasCache('informations')) {
-      res.status(200).send(cache.getCache('informations'));
+    if (cache.getCache('informations')) {
+      const results = cache.getCache('informations');
+      res.status(200).send(results);
       return;
     }
-
     Information.find({}, 'infoName infoContent visibility')
       .then((results) => {
         // cache results
         cache.setCache('informations', results);
         res.status(200).send(results);
       })
-      .catch((error) => res.status(404).send(error));
+      /* eslint-disable no-unused-vars */
+      .catch((error) => res.status(500).send(new Error('Error when finding infoName')));
   };
 
   const addInformation = function (req, res) {
-    Information.find({ infoName: { $regex: escapeRegex(req.body.infoName), $options: 'i' } })
-        .then((result) => {
-            if (result.length > 0) {
-                res.status(400).send({ error: `Info Name must be unique. Another infoName with name ${result[0].infoName} already exists. Please note that info names are case insensitive` });
-                return;
-            }
-            const _info = new Information();
-            _info.infoName = req.body.infoName;
-            _info.infoContent = req.body.infoContent || 'Unspecified';
-            _info.visibility = req.body.visibility || '0';
-            _info.save()
-                .then((newInformation) => {
-                  // remove cache if cache is available
-                  if (cache.hasCache('informations')) {
-                    cache.removeCache('informations');
-                    return;
-                  }
-                  res.status(201).send(newInformation);
-                })
-                .catch((error) => res.status(400).send(error));
-        })
-        .catch((error) => res.status(500).send({ error }));
+    try {
+      const result = Information.findOne({
```

Comment:
You changed from `Information.find` to `Information.findOne`. The first one returns an array; the second one, and object. This will break line 36 since it's accessing index 0 (this will throw an exception in integration tests).

Lastly, it may not be necessary to turn the chaining promise into try catch blocks. We have now a flushPromises function that helps jest being able to test code with chaining promises.

While I do prefer to use async await rather than chaining promises, refactoring the code could potentially lead to errors (hence why we add integration tests, but even with that, the frontend can be impacted).